src/app/containers/Article/index.jsx

```diff
@@ -3,11 +3,22 @@ import { bool, string, shape } from 'prop-types';
 import Metadata from '../../components/Metadata';
 import Header from '../../components/Header';
 import Footer from '../Footer';
-import MainContent from '../MainContent';
+import headings from '../Headings';
+import text from '../Text';
+import image from '../Image';
+import Blocks from '../Blocks';
+import MainContent from '../../components/MainContent';
 import articlePropTypes from '../../models/propTypes/article';
 import { ServiceContextProvider } from '../../components/ServiceContext';
 import serviceConfig from '../../lib/serviceConfig';
 
+const componentsToRender = {
```

Comment:
This mapping should be within the Blocks container, rather than here in the Article container. `Article` shouldn't need to care about the implementation details in a child component, `Blocks`. 

In addition to this, we should separately open an issue to add more testing around this mapping logic. The testing around such a core part of our application is minimal at the moment.

Reply:
The main point of this refactor was to do that abstraction and not tie `Blocks` to rendering specific components (which is why the name changed from `MainContent` to `Blocks`).

Through this refactor, there can be a single implementation/component for rendering blocks; `Blocks` can be used to render fragments (which is just a type of block), and it can also be used to render/map other types of blocks in the future.

Reply:
I agree that it is good to separate out the `MainContent` component - which just now handles the `<main role="main>` and grid layout `div`s  - from the `Blocks` logic. 

If you want to make `Blocks` generic enough to handle future blocks, you should also pull out the props from here https://github.com/bbc/simorgh/blob/e9548b6c66d3ded3455295c2c534e9ff249a3d7e/src/app/containers/Blocks/index.jsx#L32-L35

I do not think this mapping should live within the Article container, though. Ideally, I'd like it to be in a separate container so that it can be tested thoroughly independently of Articles container logic.

Reply:
If we're keeping the mapping within the Article container for now, we should simplify the proptypes for componentsToRender to

```
componentsToRender: shape({}).isRequired,
```

This is so that this particular component is self-contained and independent of the Article container.

Reply:
Done!